pkg/controlplane/controller.go

```diff
@@ -174,47 +174,27 @@ func (c *Controller) Start() {
 	repairClusterIPs := servicecontroller.NewRepair(c.ServiceClusterIPInterval, c.ServiceClient, c.EventClient, &c.ServiceClusterIPRange, c.ServiceClusterIPRegistry, &c.SecondaryServiceClusterIPRange, c.SecondaryServiceClusterIPRegistry)
 	repairNodePorts := portallocatorcontroller.NewRepair(c.ServiceNodePortInterval, c.ServiceClient, c.EventClient, c.ServiceNodePortRange, c.ServiceNodePortRegistry)
 
-	// We start both repairClusterIPs and repairNodePorts to catch events
-	// coming from the RunOnce() calls to them.
-	// TODO: Refactor both controllers to only expose a public RunUntil
-	// method, but to accommodate the usecase below (of forcing the first
-	// successful call before proceeding) let them signal on that, like:
-	//   func (c *Repair) RunUntil(stopCh chan struct{}, onFirstSuccess func())
```

Comment:
what about having a `RunOnceWait() <-chan struct`  instead of adding a function since we are only use this function for wait it runs once, we can use that to block and allows use to set a timeout
```
select {
case <- repairClusterIPs.RunOnceWait():
case <- time.After(15*time.Second):
       klog.Fatal()
}
```

Reply:
I personally feel that increasing the interface for the reason of somewhat artificial backward compatibility isn't worth it. I don't think we ever clearly document that we will fatal on if we can't repair even once at start.

Reply:
that is a good point